src/null-safety/faq.md

```diff
@@ -249,20 +249,33 @@ use the [`filled`](https://api.dart.dev/stable/dart-core/List/List.filled.html)
 
 {:.bad}
 {% prettify dart tag=pre+code %}
-_jellyPoints = List<Vec2D>(jellyMax + 1);
+_jellyCounts = List<int?>(jellyMax + 1);
 for (var i = 0; i <= jellyMax; i++) {
-  _jellyPoints[i] = Vec2D(); // List initalized with the same value
+  _jellyCounts[i] = 0; // List initalized with the same value
 }
 {% endprettify %}
 
 {:.good}
 {% prettify dart tag=pre+code %}
-_jellyPoints = List<Vec2D>.filled(jellyMax + 1, Vec2D()); // List initialized with filled constructor
+_jellyCounts = List<int>.filled(jellyMax + 1, 0); // List initialized with filled constructor
 {% endprettify %}
 
-If you are setting the elements of the list via an index, you should instead use
-the `add` function to build the list. That is less error-prone and more
-readable.
+If you are setting the elements of the list via an index, or you are populating
+each element of the list with a distinct value, you should instead use the
+[`generate`](https://api.dart.dev/stable/dart-core/List/List.generate.html) constructor to build the list.
+
+{:.bad}
+{% prettify dart tag=pre+code %}
+_jellyPoints = List<Vec2D?>(jellyMax + 1);
+for (var i = 0; i <= jellyMax; i++) {
```

Comment:
[nit] The rewrite I would do is

```dart
_jellyPoints = [
  for (var i = 0; i <= jellyMax; i++)
    Vec2D(),
 ];
 ```
 
 We could make this slightly more realistic if we add a `growable: false` argument to `List.generate` which makes it infeasible to use the list literal. I'm not sure if doing that would add noise that detracts from the point though?

Reply:
Honestly that's the rewrite I would have done too.  But I have the impression that `List.generate` is what we're trying to steer users to these days (and that the VM and dart2js folks have put special effort into making sure it's performant), so I figured 
I would go with that.  @leafpetersen / @munificent, do either of you have a preference for what we recommend here?

Reply:
List.generate was optimized: https://github.com/dart-lang/sdk/issues/42283 . Not sure about the collection literal, maybe @alexmarkov , @mraleph or @rakudrama can comment here.  In general though, I'd probably steer people first and foremost towards the most maintainable/readable code (so long as perf is adequate).  So my vote is probably for the collection literal.

Reply:
Ok, switched to a collection literal.